fdbclient/BlobStore.actor.cpp

```diff
@@ -598,7 +616,9 @@ ACTOR Future<Reference<HTTP::Response>> doRequest_impl(Reference<BlobStoreEndpoi
 			// Finish/update the request headers (which includes Date header)
 			// This must be done AFTER the connection is ready because if credentials are coming from disk they are refreshed
 			// when a new connection is established and setAuthHeaders() would need the updated secret.
-			bstore->setAuthHeaders(verb, resource, headers);
+			if (bstore->credentials.present()) {
```

Comment:
Since `credentials` is a member of `BlobStoreEndpoint` and `setAuthHeaders()` is a non-static member function of `BlobStoreEndpoint`, couldn't `setAuthHeaders()` be called unconditionally, and internally it checks for `credentials.present()` and returns after doing nothing unless it is present?

Similarly, `hmac_sha1()` is a member function of the same class so it doesn't need the `credentials` argument to be passed in either.

Reply:
Thank you for the review!

The motivation of including `credentials` as a parameter to these functions is to guarantee its existence. I agree that `setAuthHeaders` could do nothing if there are no credentials, but what should the behavior of `hmac_sha1` be if the credentials do not exist?

In looking closer at `hmac_sha1`, it doesn't use any member variables (besides `credentials`). Perhaps we could make `hmac_sha1` a static method?

Reply:
You're right that `hmac_sha1()` should be static, there's no point in calling it without an account input.

I see `setAuthHeaders()` logically as a request to "Set the required auth headers."  In the case where there are no account credentials, there is nothing to set, so I think doing nothing if the optional is not present is valid logic.